langchain_chat.py

```diff
@@ -0,0 +1,75 @@
+import streamlit as st
+from streamlit_chat import message
+from langchain.embeddings.openai import OpenAIEmbeddings
+from langchain.chat_models import ChatOpenAI
+from langchain.chains import ConversationalRetrievalChain
+from langchain.document_loaders.csv_loader import CSVLoader
+from langchain.vectorstores import FAISS
+import tempfile
+
+user_api_key = st.sidebar.text_input(
+    label="#### Your OpenAI API key 👇",
+    placeholder="Paste your openAI API key, sk-",
+    type="password")
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The OpenAI API key is sensitive information and should not be handled in plaintext. Consider using a more secure method to handle this, such as environment variables or secure vaults.
```diff
- user_api_key = st.sidebar.text_input(
-     label="#### Your OpenAI API key 👇",
-     placeholder="Paste your openAI API key, sk-",
-     type="password")
+ import os
+ user_api_key = os.getenv('OPENAI_API_KEY')
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->